Tests/PackageGraphTests/PubgrubTests.swift

```diff
@@ -1289,17 +1293,19 @@ final class PubGrubDiagnosticsTests: XCTestCase {
             "foo": (.versionSet(v1Range), .specific(["foo"])),
         ])
         let result = resolver.solve(dependencies: dependencies)
-
+        
+        print(result.errorMsg!)
+        
         XCTAssertEqual(result.errorMsg, """
-          because every version of a[a] depends on b[b] 2.0.0..<3.0.0 and foo[foo] <1.1.0 depends on a[a] 1.0.0..<2.0.0, foo[foo] <1.1.0 requires b[b] 2.0.0..<3.0.0.
-             (1) So, because foo[foo] <1.1.0 depends on b[b] 1.0.0..<2.0.0, foo[foo] <1.1.0 is forbidden.
-          because every version of x[x] depends on y[y] 2.0.0..<3.0.0 and foo[foo] >=1.1.0 depends on x[x] 1.0.0..<2.0.0, foo[foo] >=1.1.0 requires y[y] 2.0.0..<3.0.0.
-          And because foo[foo] >=1.1.0 depends on y[y] 1.0.0..<2.0.0, foo[foo] >=1.1.0 is forbidden.
-          And because foo[foo] <1.1.0 is forbidden (1), foo[foo] is forbidden.
-          And because root depends on foo[foo] 1.0.0..<2.0.0, version solving failed.
+          Dependencies could not be resolved because root depends on 'foo' 1.0.0..<2.0.0.
+          'foo' cannot be used because 'foo' < 1.1.0 cannot be used (1).
+          'foo' >= 1.1.0 cannot be used because 'foo' >= 1.1.0 depends on 'y' 1.0.0..<2.0.0.
+          'foo' >= 1.1.0 practically depends on 'y' 2.0.0..<3.0.0 because 'x' depends on 'y' 2.0.0..<3.0.0 and 'foo' >= 1.1.0 depends on 'x' 1.0.0..<2.0.0.
+          'foo' < 1.1.0 practically depends on 'b' 2.0.0..<3.0.0 because 'a' depends on 'b' 2.0.0..<3.0.0 and 'foo' < 1.1.0 depends on 'a' 1.0.0..<2.0.0.
+             (1) As a result, 'foo' < 1.1.0 cannot be used because 'foo' < 1.1.0 depends on 'b' 1.0.0..<2.0.0.
```

Comment:
Does this indicate a footnote to the second line of the error message?  If so, I'll think about a way in which this could be annotated in the DiagnosticsEngine so that an IDE could display it indented under that line.

Reply:
yes